src/components/current-show/CurrentShow.js

```diff
@@ -54,60 +54,58 @@ const CurrentShow = ({ currentShow, residentsData }) => {
       return sanitiseString(prismicShow.data.show_description);
     }
-    if (currentShow !== null) {
-      currentShowDescription = currentShow.description;
-      if (currentShowDescription === "") {
-        currentShowDescription = "Edinburgh Community Radio.";
-        return currentShowDescription;
-      } else {
-        return sanitiseString(currentShowDescription);
-      }
+
+    if (currentShow && currentShow.description) {
+      return sanitiseString(currentShow.description);
     }
+
+    return "Edinburgh Community Radio.";
   };
 
   return (
     <Wrapper>
       <OnAirWrapper>
-        <OnAir />
+        <OnAir on={Boolean(currentShow)} />
       </OnAirWrapper>
-      {prismicShow && (
-        <>
-          <CurrentShowImageWrapper>{returnImage()}</CurrentShowImageWrapper>
-          <InfoWrapper>
-            {prismicShow === SHOW_NOT_FOUND ? (
-              <NameWrapper>
-                <ShowName>{parseShowName(currentShow)}</ShowName>
-              </NameWrapper>
-            ) : (
-              <Link to={`/residents/${prismicShow.uid}`}>
-                <NameWrapper
-                  onMouseOver={() => {
-                    setHovered(true);
-                  }}
-                  onMouseOut={() => {
-                    setHovered(false);
-                  }}
-                >
-                  <ShowName>{parseShowName(currentShow)}</ShowName>
-                  <HoveredLine
-                    hovered={hovered}
-                    width={"100%"}
-                    placeholderWidth={"3rem"}
-                    placeholder
-                  />
-                </NameWrapper>
-              </Link>
-            )}
-            <DescriptionWrapper>
-              <ShowDescription>{returnShowDescription()}</ShowDescription>
-            </DescriptionWrapper>
-          </InfoWrapper>
-        </>
-      )}
+      <CurrentShowImageWrapper>{returnImage()}</CurrentShowImageWrapper>
+      <InfoWrapper>
+        {currentShow && prismicShow === SHOW_NOT_FOUND ? (
+          <NameWrapper>
+            <ShowName>{parseShowName(currentShow)}</ShowName>
+          </NameWrapper>
+        ) : null}
+        {currentShow && prismicShow && prismicShow.uid ? (
+          <Link to={`/residents/${(prismicShow && prismicShow.uid) || ""}`}>
+            <NameWrapper
+              onMouseOver={() => {
+                setHovered(true);
+              }}
+              onMouseOut={() => {
+                setHovered(false);
+              }}
+            >
+              <ShowName>{parseShowName(currentShow)}</ShowName>
+              <HoveredLine
+                hovered={hovered}
+                width={"100%"}
+                placeholderWidth={"3rem"}
+                placeholder
+              />
+            </NameWrapper>
+          </Link>
+        ) : null}
+        {!currentShow ? (
+          <NameWrapper>
+            <ShowName>We are currently offline.</ShowName>
+          </NameWrapper>
+        ) : null}
```

Comment:
Are ternaries necessary for all these? `{ !currentShow &&... ` isn't enough?

Reply:
There's [a reason not to use](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) the `&&` for component rendering logic. And it's the few random cases when `!someData` evaluate in 0, and like that you end up with a screen with rogue zero :D